core/src/main/java/org/elasticsearch/discovery/Discovery.java

```diff
@@ -87,4 +88,10 @@ public FailedToCommitClusterStateException(String msg, Throwable cause, Object..
             super(msg, cause, args);
         }
     }
+
+    /**
+     * @return stats about the discovery
+     */
+    PendingClusterStateStats stats();
```

Comment:
And, I dont't know if I understand this correctly. But look at `NodeStats.java` line 228, I think use a interface here would be really difficult here.